packages/components/src/components/amount-input/amount-input.tsx

```diff
@@ -39,6 +39,10 @@ const AmountInput = ({
         [decimal_places, locale]
     );
 
+    useEffect(() => {
+        setValue(initial_value);
+    }, [initial_value]);
```

Comment:
```suggestion
    useEffect(() => {
        onChangeHandler({ target: { value: initial_value.toString() } });
    }, [initial_value, onChangeHandler]);
```
@heorhi-deriv It should call `onChangeHandler` to make sure all the check happens, @rostislav-deriv please correct me if I'm wrong 🙏🏻

Reply:
@farzin-deriv since we will replace onChangeHandler with hooks sooner, i leave the comment here, because current improvement takes global refactoring of this component